app/controllers/catalog_controller.rb

```diff
@@ -915,6 +919,24 @@ def ansible_playbook?
   end
   helper_method :ansible_playbook?
 
+  def terraform_template_type?
+    prov_type = if params[:st_prov_type]
+                  params[:st_prov_type]
+                elsif @record
+                  @record.prov_type
+                elsif @edit
+                  @edit[:new][:st_prov_type]
+                end
+    prov_type == 'generic_terraform_template'
+  end
+
+  def terraform_template?
+    terraform_template = terraform_template_type?
+    @current_region = MiqRegion.my_region.region if terraform_template
+    terraform_template
```

Comment:
This is weird, but I see it is just copied from the ansible_playbook method above.  Seems like we could handle setting @current_region separately from checking terraform_template_type? then we don't need this extra method